build/shimplify.py

```diff
@@ -392,6 +392,11 @@ def __traverse_source_tree_of_all_shims(src_type, func):
                     "orphan shim files should be deleted"
                 build_ver_arr = map(lambda s: str(json.loads(s).get('spark')), shim_arr)
                 __log.debug("extracted shims %s", build_ver_arr)
+                for ver in build_ver_arr:
```

Comment:
instead of making an extra loop we could create a function for map on L393
that after extracting version would check against 
```Python
shim_ver = str(json.loads(s).get('spark'))
assert shim_ver in __all_shims_arr, "all.buildvers in pom.xml does not contain %s" % shim_ver
```